src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -406,13 +403,11 @@ else if (node.getType() == JavadocTokenTypes.HTML_ELEMENT
      * @return {@code true} if first tag is summary tag.
      */
     private static boolean isSummaryTag(DetailNode javadoc) {
-        final DetailNode[] child = javadoc.getChildren();
-
-        // Checking size of ast is not required, since ast contains
-        // children of Inline Tag, as at least 2 children will be present which are
-        // RCURLY and LCURLY.
-        return child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
-                && SUMMARY_TEXT.equals(child[1].getText());
+        return Arrays.stream(javadoc.getChildren())
+                .anyMatch(child -> {
```

Comment:
Using `anyMatch()`, with Java streams gives no guarantee regarding the order of traversal,  this method could now return true if ANY tag is a summary tag, not just the first.

Reply:
I don't think `CUSTOME_NAME` or rather any other secondary tag can be present in children except the first one because our grammar defines only the first and tag rest are considered text or description. What do you think?
```
javadocInlineTag:
      JAVADOC_INLINE_TAG_START
      (
            CODE_LITERAL (WS | NEWLINE | LEADING_ASTERISK | text)*
            | DOC_ROOT_LITERAL (WS | NEWLINE | LEADING_ASTERISK)*
            | INHERIT_DOC_LITERAL (WS | NEWLINE | LEADING_ASTERISK)*
            | LINK_LITERAL (WS | NEWLINE | LEADING_ASTERISK)+ reference (WS | NEWLINE)*
                ((WS | NEWLINE) description)?
            | LINKPLAIN_LITERAL (WS | NEWLINE | LEADING_ASTERISK)+ reference (WS | NEWLINE)*
                ((WS | NEWLINE) description)?
            | LITERAL_LITERAL (WS | NEWLINE | LEADING_ASTERISK | text)*
            | VALUE_LITERAL (WS | NEWLINE | LEADING_ASTERISK)* ((WS | NEWLINE) reference)?
            | CUSTOM_NAME (WS | NEWLINE | LEADING_ASTERISK)* ((WS | NEWLINE) description)?
      )
      JAVADOC_INLINE_TAG_END
      ;
```

Reply:
> What do you think?

I think that the javadoc states that this method will `@return {@code true} if first tag is summary tag.` , but that your changes no longer guarantee that :)

Reply:
Can you please explain in a bit? In what cases it can fail? Since we know that there will only be a single `JAVADOC_INLINE_TAG` present, how would a case exits where the type is `CUSTOM_NAME`.

Reply:
> I just mentioned this because it is in the javadoc description, and the old code made it very apparent that we were getting the first element in the array.

True, it was visually clear in the previous code, but I thought to make it a bit redundant by not hard-coding things out. But the javadoc still ought to remain same as we still are directly checking the first `INLINE_JAVADOC_TAG`.

Reply:
> I just mentioned this because it is in the javadoc description, and the old code made it very apparent that we were getting the first element in the array.

True, it was visually clear in the previous code, but I thought to make it a bit redundant by not hard-coding things out. But the javadoc still ought to remain same as we still are directly checking the first INLINE_JAVADOC_TAG. Please resolve things out if you think the problem is solved.

Reply:
> In what cases it can fail?

In practice, probably never :)

What I am saying is that this sort of stream usage is probably overkill, and using a non-deterministic API to get something that is supposed to be in an exact index seems like a bad idea to me. If you want to clean up the code a bit, why not just do:

```java

        final DetailNode firstChild = javadoc.getChildren()[1];
        return firstChild.getType() == JavadocTokenTypes.CUSTOM_NAME
                && SUMMARY_TEXT.equals(firstChild.getText());

```
or

```java
        final DetailNode firstChild = Arrays.asList(javadoc.getChildren()).get(1);
        return firstChild.getType() == JavadocTokenTypes.CUSTOM_NAME
                && SUMMARY_TEXT.equals(firstChild.getText());

```

?

This 2-3 lines of code is concise, easy to debug, and makes the intention very clear. I do not think that this stream usage is beneficial here.

Reply:
Let us not change it then, as it is not a big change.

Reply:
Resolved.